File: backend/internal/domain/group.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type PaymentStatus string

const (
	PaymentStatusTrial   PaymentStatus = "trial"
	PaymentStatusPending PaymentStatus = "pending"
	PaymentStatusPaid    PaymentStatus = "paid"
)

type Group struct {
	ID          uuid.UUID  `db:"id"`
	TeacherID   uuid.UUID  `db:"teacher_id"`
	Name        string     `db:"name"`
	Subject     *string    `db:"subject"`
	Description *string    `db:"description"`
	InviteCode  string     `db:"invite_code"`
	IsArchived  bool       `db:"is_archived"`
	CreatedAt   time.Time  `db:"created_at"`
	UpdatedAt   time.Time  `db:"updated_at"`
	DeletedAt   *time.Time `db:"deleted_at"`
}

type GroupMember struct {
	GroupID       uuid.UUID     `db:"group_id"`
	StudentID     uuid.UUID     `db:"student_id"`
	JoinedAt      time.Time     `db:"joined_at"`
	PaymentStatus PaymentStatus `db:"payment_status"`
}

// GroupMemberDetail — GroupMember + данные пользователя (для списка участников).
type GroupMemberDetail struct {
	GroupMember
	Name      string  `db:"name"`
	Phone     string  `db:"phone"`
	AvatarURL *string `db:"avatar_url"`
}

File: flutter_app/lib/features/groups/domain/entities/group.dart
import 'package:flutter/foundation.dart';

enum PaymentStatus {
  trial,
  pending,
  paid;

  static PaymentStatus fromJson(String v) => switch (v) {
        'paid' => paid,
        'pending' => pending,
        _ => trial,
      };

  String toJson() => name;
}

@immutable
class Group {
  const Group({
    required this.id,
    required this.teacherId,
    required this.name,
    required this.inviteCode,
    required this.isArchived,
    required this.createdAt,
    this.subject,
    this.description,
    this.memberCount = 0,
  });

  final String id;
  final String teacherId;
  final String name;
  final String? subject;
  final String? description;
  final String inviteCode;
  final bool isArchived;
  final int memberCount;
  final DateTime createdAt;

  Group copyWith({
    String? name,
    String? subject,
    String? description,
    bool? isArchived,
    int? memberCount,
  }) =>
      Group(
        id: id,
        teacherId: teacherId,
        name: name ?? this.name,
        subject: subject ?? this.subject,
        description: description ?? this.description,
        inviteCode: inviteCode,
        isArchived: isArchived ?? this.isArchived,
        memberCount: memberCount ?? this.memberCount,
        createdAt: createdAt,
      );
}

@immutable
class GroupMember {
  const GroupMember({
    required this.studentId,
    required this.name,
    required this.phone,
    required this.joinedAt,
    required this.paymentStatus,
    this.avatarUrl,
  });

  final String studentId;
  final String name;
  final String phone;
  final String? avatarUrl;
  final DateTime joinedAt;
  final PaymentStatus paymentStatus;
}
